pkg/ingester/transfer.go

```diff
@@ -35,25 +41,20 @@ func init() {
 // TransferChunks receives all the chunks from another ingester.
 func (i *Ingester) TransferChunks(stream client.Ingester_TransferChunksServer) error {
 	// Enter JOINING state (only valid from PENDING)
-	if err := i.ChangeState(ring.JOINING); err != nil {
+	if err := i.lifecycler.ChangeState(ring.JOINING); err != nil {
 		return err
 	}
 
 	// The ingesters state effectively works as a giant mutex around this whole
 	// method, and as such we have to ensure we unlock the mutex.
 	defer func() {
-		errc := make(chan error)
-		i.actorChan <- func() {
-			if i.getState() == ring.JOINING {
-				errc <- i.changeState(ring.PENDING)
-			} else {
-				errc <- nil
+		if i.lifecycler.GetState() != ring.ACTIVE {
+			// Enter PENDING state (only valid from JOINING)
```

Comment:
Given the bracketed part of this comment, would it be better to check the state and have a separate error message for the condition that we have entered a state that is not ACTIVE and not JOINING ?

(This is a behaviour change; previously it would have sailed on without error)